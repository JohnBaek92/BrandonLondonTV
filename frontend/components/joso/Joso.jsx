import React from "react";

class Joso extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      recentJOSO: [],
      videos: []
    };

    this.onSearchResponse = this.onSearchResponse.bind(this);
    this.onYouTubeApiLoad = this.onYouTubeApiLoad.bind(this);
    this.renderVideos = this.renderVideos.bind(this);
  }

  componentWillMount() {
    window.gapi.load("client", () => {
      window.gapi.client.load("youtube", "v3", this.onYouTubeApiLoad);
    });
  }

  onYouTubeApiLoad() {
    window.gapi.client.setApiKey("AIzaSyD7LR5NLIRrMUB6IrmJa0MPpVInKGQiGdA");
    this.search();
  }

  search() {
    const request = window.gapi.client.youtube.playlistItems.list({
      part: "snippet",
      playlistId: "PLqtmVPBck9G7ZoZ_4eEJ8tl_EoFlETZho",
      order: "date",
      maxResults: 2
    });
    request.execute(this.onSearchResponse);
  }

  onSearchResponse(response) {
    const videos = response.items;
    this.setState({ videos });
  }

  renderVideos() {
    const videos = this.state.videos;
    return videos.map((video, idx) => (
      <iframe
        key={idx}
        width="560"
        height="400"
        src={`https://www.youtube.com/embed/${
          video.snippet.resourceId.videoId
        }`}
        frameBorder="0"
        allow="autoplay; encrypted-media"
        allowFullScreen
      />
    ));
  }


  render() {
    return (
      <div>
        <div className="title is-size-2 has-text-weight-bold has-text-centered">
          When The Game Ends, The Lifestyle Begins
        </div>
        <div className="columns is-mobile">
          <div className="column sc-joso is-two-thirds centered">
            <iframe
              width="100%"
              height="166"
              scrolling="no"
              frameBorder="no"
              allow="autoplay"
              src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/392031732&amp;color=%23FFD046&amp;auto_play=false&amp;hide_related=false&amp;show_comments=false&amp;show_user=true&amp;show_reposts=false&amp;show_teaser=true"
            />
            <iframe
              width="100%"
              height="166"
              scrolling="no"
              frameBorder="no"
              allow="autoplay"
              src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/388488576&amp;color=%23FFD046&amp;auto_play=false&amp;hide_related=false&amp;show_comments=false&amp;show_user=true&amp;show_reposts=false&amp;show_teaser=true"
            />
          </div>
        </div>
      </div>
    );
  }
}

export default Joso;
