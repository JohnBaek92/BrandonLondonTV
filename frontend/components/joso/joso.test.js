import React from "react";
import { render } from "enzyme";
import toJson from "enzyme-to-json";
import Joso from "./Joso";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

test("Joso component should render as expected", () => {
  const component = render(<Joso />);
  const tree = toJson(component);

  it("Loads the Youtube API", () => {
    expect(global.window.gapi.load).toBeCalled();
  });
});
