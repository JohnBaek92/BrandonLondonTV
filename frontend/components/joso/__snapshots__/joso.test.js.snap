// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Joso component should render as expected Renders the Joso correctly 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "0s5v0l",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "0s5v0l",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Joso>
      <div>
        <div
          className="title is-size-2 has-text-weight-bold has-text-centered"
        >
          When The Game Ends, The Lifestyle Begins
        </div>
        <div
          className="columns is-mobile"
        >
          <div
            className="column sc-joso is-two-thirds centered"
          >
            <iframe
              allow="autoplay"
              frameBorder="no"
              height="166"
              scrolling="no"
              src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/392031732&color=%23FFD046&auto_play=false&hide_related=false&show_comments=false&show_user=true&show_reposts=false&show_teaser=true"
              width="100%"
            />
            <iframe
              allow="autoplay"
              frameBorder="no"
              height="166"
              scrolling="no"
              src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/388488576&color=%23FFD046&auto_play=false&hide_related=false&show_comments=false&show_user=true&show_reposts=false&show_teaser=true"
              width="100%"
            />
          </div>
        </div>
      </div>
    </Joso>
  </Router>
</MemoryRouter>
`;
